apps/condo/domains/organization/sbbol/routes.js

```diff
@@ -0,0 +1,60 @@
+const { generators } = require('openid-client') // certified openid client will all checks
+const { SbbolUserInfoJSONValidation, SBBOL_SESSION_KEY } = require('@condo/domains/organization/sbbol/common')
+const { SbbolOauth2Api } = require('@condo/domains/organization/sbbol/oauth2')
+const { SbbolOrganization } = require('@condo/domains/organization/sbbol/synch')
+const { JSON_SCHEMA_VALIDATION_ERROR } = require('@condo/domains/common/constants/errors')
+const { getSchemaCtx } = require('@core/keystone/schema')
+
+class SbbolRoutes {
+    constructor () {
+        this.helper = new SbbolOauth2Api()
+    }
+
+    startAuth () {
```

Comment:
async startAuth (req, res, next)  {
        // nonce: to prevent several callbacks from same request
        // state: to validate user browser on callback
        const checks = { nonce: generators.nonce(), state: generators.state() }
        req.session[SBBOL_SESSION_KEY] = checks
        await req.session.save()
        try {
            const redirectUrl = this.helper.authorizationUrlWithParams(checks)
            return res.redirect(redirectUrl)
        } catch (error) {
            return next(error)
        }
    }

Reply:
fixed